StorageTests/TestSQLiteHistory.swift

```diff
@@ -997,13 +998,13 @@ class TestSQLiteHistory: XCTestCase {
             return history.insertOrUpdatePlace(site13, modified: Date.nowMicroseconds())
         }).bind({ guid in
             XCTAssertEqual(guid.successValue!, initialGuid, "Guid is correct")
-            return history.getSitesByFrecencyWithHistoryLimit(10)
+            return history.getFrecentHistory().getSites(historyLimit: 10, bookmarksLimit: 0, whereURLContains: nil)
         }).bind({ (sites: Maybe<Cursor<Site>>) -> Success in
             XCTAssert(sites.successValue!.count == 2, "2 sites returned")
```

Comment:
This test fails, and the test itself makes no sense to me what it is trying to do, making it hard to fix.
It appears to expect frecency to coalesce on ETLD, which it doesn't. This looks like if should be a top sites query and not frecency because later it removes from top sites, and re-queries frecency expecting it to change.

Reply:
Prior to my commit, history.getSitesByFrecencyWithHistoryLimit(10) will abort (complains about a missing history.is_deleted) if a bookmark limit is set to non-zero and no url filter passed in.
New code has the same bug.

Reply:
It's conceivable that the test bitrotted in a way that didn't cause it to fail; this code has evolved constantly for years. If the behavior makes no sense when both parameters are passed, perhaps `gSBFWHL` should have a nice informative `assert` at the top of the method?

Reply:
I would also need to re-introduce the 'side-effect' that depending on which overload one calls, the function behaves very differently. Not a big of fan of that, although if I am feeling desperate that may be the simplest approach. 
My first thought on this is to try and clean up the bitrot, and that this test may be testing a use-case that is never called by the app, essentially doing nothing more than verifying this side-effect of a particular overload.